api/src/main/kotlin/com/openlattice/search/Auth0UserSearchFields.kt

```diff
@@ -0,0 +1,20 @@
+package com.openlattice.search
+
+import com.fasterxml.jackson.annotation.JsonProperty
+
+// https://auth0.com/docs/users/user-search/user-search-query-syntax
+data class Auth0UserSearchFields(
+    @JsonProperty val email: String?,
+    @JsonProperty val name: String?
+) {
+    init {
+        // TODO - support multiple fields and construct a valid Lucene query string to pass to Auth0
+        require((email != null && name != null).not()) { "only one of \"email\", \"name\" are allowed" }
```

Comment:
I think there's a missing assertion here that at least one of them must be present. I think it's a bit easier to read this as
```
require(email != null || name != null) { "<can't both be null, blah>" } 
require(email == null || name == null) { "<can't both be present, blah>" }

...not blank assertions...
```

or! if you want to be super fancy and do all these constrictor assertion in one line, you could do:
```
require(arrayOf(name, email).filterNot { it.isNullOrBlank() }.size == 1) {
  <exactly one of email, name must be present and not blank, blah>
}
```

(although the former is way more intuitive so I prefer it for that reason)

Reply:
I just caught this bug too. I'll upate.